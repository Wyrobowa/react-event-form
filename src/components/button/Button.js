import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';

const Button = forwardRef(({
  type, onClick, text, className,
}, ref) => (
  // eslint-disable-next-line react/button-has-type
  <button type={type} onClick={onClick} className={className} ref={ref}>
    {text}
  </button>
));

Button.propTypes = {
  type: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  text: PropTypes.string,
  className: PropTypes.string,
};

Button.defaultProps = {
  type: 'button',
  text: '',
  className: 'btn',
};

export default Button;
