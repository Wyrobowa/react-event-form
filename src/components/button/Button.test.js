import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button', () => {
  it('renders Button component without crashing', () => {
    const component = shallow(<Button type="button" text="Button text" onClick={() => {}} />);

    expect(component).toMatchSnapshot();
  });
});
