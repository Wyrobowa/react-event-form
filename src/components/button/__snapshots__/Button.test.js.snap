// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders Button component without crashing 1`] = `
<button
  className="btn"
  onClick={[Function]}
  type="button"
>
  Button text
</button>
`;
